### ISSUE TITLE
Feature idea for E2, Templates for fresh E2s

### ISSUE THREAD

**Issue by Tjackjesus** [wiremod/wire#3176]
I was told to go here for this feature idea after i asked about it on the discord!

Its a simple idea really, its just the ability to have templates open, so you dont always start with the basic "@name @inputs" etc, but instead you can choose freely, so i.e i'd choose to start with "@name @inputs W A S D if(first()|dupefinished()) {} " and what not, it feels like a useful idea, because i rarely dont use the if first statement, so it'd be nice to have it always there :)

**Comment by deltamolfar** [comment]
Want to do that, so I've asked him to create an issue here, so it doesn't get lost.
Basic implementation idea (once I'm done with popup lib):
New auto-gen file: `_templates/default.txt` to where the current default e2 content goes.

On new e2 created - if there is more then 1 file in _templates > dialog opens (that can be blocked, and then unblocked from settings) that prompts user to selected template to choose from.

Also examples and tutorials can go there in `_tutorials` or whatever subfolder

**Comment by Denneisk** [comment]
Yes, yes, and with templates we can stop people from using `if(first())` and force `@trigger none`...

**Comment by Tjackjesus** [comment]
> Yes, yes, and with templates we can stop people from using `if(first())` and enforce `@trigger none`...

I feel targeted by this comment... ( ͡° ͜ʖ ͡°)

**Comment by Denneisk** [comment]
> On new e2 created - if there is more then 1 file in _templates > dialog opens (that can be blocked, and then unblocked from settings) that prompts user to selected template to choose from.

I don't know how viable it is with the setup we have, but I feel like ideally templates would be loaded by a hyperlink in the editor.

Maybe instead we could make a landing page? Meh.

<Details><Summary>Landing page example</Summary>

<img src="https://winaero.com/blog/wp-content/uploads/2017/02/1_9_welcome-page.png"/>

</Details>

Either of these suggestions make me want to try and see how bad an HTML E2 editor could be.

**Comment by deltamolfar** [comment]
``Maybe instead we could make a landing page? Meh.``
Such page would be nice on first opened file. But not when you want to create a new file, that's for sure.

**Comment by Vurv78** [comment]
"not a bug" isnt used for suggestions. convert it to a discussion

**Comment by Denneisk** [comment]
Oh yeah
